c-api/src/sec_handle.rs

```diff
@@ -700,19 +700,61 @@ pub type SetCredentialsAttributesFnW =
 
 #[cfg_attr(windows, rename_symbol(to = "Rust_ChangeAccountPasswordA"))]
 #[no_mangle]
-pub extern "system" fn ChangeAccountPasswordA(
-    _psz_package_name: *mut SecChar,
-    _psz_domain_name: *mut SecChar,
-    _psz_account_name: *mut SecChar,
-    _psz_old_password: *mut SecChar,
-    _psz_new_password: *mut SecChar,
+pub unsafe extern "system" fn ChangeAccountPasswordA(
+    psz_package_name: *mut SecChar,
+    psz_domain_name: *mut SecChar,
+    psz_account_name: *mut SecChar,
+    psz_old_password: *mut SecChar,
+    psz_new_password: *mut SecChar,
     _b_impersonating: bool,
     _dw_reserved: c_ulong,
-    _p_output: PSecBufferDesc,
+    p_output: PSecBufferDesc,
 ) -> SecurityStatus {
-    ErrorKind::UnsupportedFunction.to_u32().unwrap()
+    check_null!(psz_package_name);
+    check_null!(psz_domain_name);
+    check_null!(psz_account_name);
+    check_null!(psz_old_password);
+    check_null!(psz_new_password);
+    check_null!(p_output);
+
+    let security_package_name = CStr::from_ptr(psz_package_name).to_str().unwrap();
+
+    let domain = CStr::from_ptr(psz_domain_name).to_str().unwrap();
+    let username = CStr::from_ptr(psz_account_name).to_str().unwrap();
+    let password = CStr::from_ptr(psz_old_password).to_str().unwrap();
+    let new_password = CStr::from_ptr(psz_new_password).to_str().unwrap();
```

Comment:
I think it’s best to check the error instead of panicking here. I assume that most of the time the input will be ASCII and therefore valid UTF-8, but just to be safe. Unwinding a panic across FFI boundaries is pretty bad, if it were to happen.
You can probably use a macro like we used to have in picky c-api to make that more ergonomic: https://github.com/Devolutions/picky-rs/blob/eea7fa2d89dea68ca70b508933dfd75617375e34/c-api/src/helper.rs#L18
Make me think that you could maybe have a look at the `ffi_helpers` crate as well: https://github.com/Michael-F-Bryan/ffi_helpers

Reply:
I got rid of the `unwrap`s. Also, I added a panic catch in all the exported functions via the `check_panic!` macro(similar to one from `ffi_helpers`).  
It should be safer now 🙂